components/camera/align/join.go

```diff
@@ -0,0 +1,156 @@
+			if err != nil {
+				return nil, err
+			}
+			result, ok := attrs.(*joinAttrs)
+			if !ok {
+				return nil, rdkutils.NewUnexpectedTypeError(result, attrs)
+			}
+			return result, nil
+		}, &joinAttrs{})
+}
+
+// joinAttrs is the attribute struct for aligning.
+type joinAttrs struct {
+	ImageType            string                             `json:"output_image_type"`
+	Debug                bool                               `json:"debug,omitempty"`
+	Color                string                             `json:"color_camera_name"`
+	Depth                string                             `json:"depth_camera_name"`
+	CameraParameters     *transform.PinholeCameraIntrinsics `json:"intrinsic_parameters,omitempty"`
+	DistortionParameters *transform.BrownConrady            `json:"distortion_parameters,omitempty"`
+}
+
+func (cfg *joinAttrs) Validate(path string) ([]string, error) {
+	var deps []string
+	if cfg.Color == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "color_camera_name")
+	}
+	deps = append(deps, cfg.Color)
+	if cfg.Depth == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "depth_camera_name")
+	}
+	deps = append(deps, cfg.Depth)
+	return deps, nil
+}
+
+// joinColorDepth takes a color and depth image source and aligns them together.
+type joinColorDepth struct {
+	color, depth         gostream.VideoStream
+	colorName, depthName string
+	projector            transform.Projector
+	imageType            camera.ImageType
+	debug                bool
+	logger               golog.Logger
+}
+
+// newJoinColorDepth creates a gostream.VideoSource that aligned color and depth channels.
+func newJoinColorDepth(ctx context.Context, color, depth camera.Camera, attrs *joinAttrs, logger golog.Logger,
+) (camera.Camera, error) {
+	if attrs.CameraParameters == nil {
+		return nil, errors.Wrap(transform.ErrNoIntrinsics, "intrinsic_parameters field in attributes cannot be empty")
+	}
+	imgType := camera.ImageType(attrs.ImageType)
+	videoSrc := &joinColorDepth{
+		color:     gostream.NewEmbeddedVideoStream(color),
+		colorName: attrs.Color,
+		depth:     gostream.NewEmbeddedVideoStream(depth),
+		depthName: attrs.Depth,
+		projector: attrs.CameraParameters,
+		imageType: imgType,
+		debug:     attrs.Debug,
+		logger:    logger,
+	}
+	return camera.NewFromReader(
+		ctx,
+		videoSrc,
+		&transform.PinholeCameraModel{attrs.CameraParameters, attrs.DistortionParameters},
+		imgType,
+	)
+}
+
+// Read returns the next image from either the color or depth camera..
+// imageType parameter will determine which channel gets streamed.
+func (jcd *joinColorDepth) Read(ctx context.Context) (image.Image, func(), error) {
+	ctx, span := trace.StartSpan(ctx, "align::joinColorDepth::Read")
+	defer span.End()
+	switch jcd.imageType {
+	case camera.ColorStream, camera.UnspecifiedStream:
+		return jcd.color.Next(ctx)
+	case camera.DepthStream:
+		return jcd.depth.Next(ctx)
```

Comment:
Yes, since join_color_depth does a simple overlay of the color and depth images with no modification of either image, you can assume the depth camera will return a DepthMap by default.